grammar.js

```diff
@@ -1828,6 +1828,17 @@ module.exports = grammar({
     private: $ => 'private',
     protected: $ => 'protected',
 
+    // Represents a macro call prefixed with private/protected, e.g.
+    //   private getter foo : String
+    // This rule is separated from `call` so the `call_visibility` precedence can be
+    // applied in a targeted way.
+    call_with_visibility: $ => prec('call_visibility',
+      seq(
+        field('visibility', choice($.private, $.protected)),
+        alias($.call, 'macro call'),
```

Comment:
Using `alias($.call_with_visibility, $.call)` combined with this alias basically allows us to mix in the visibility field into the `call` node in the final tree. But `alias($.call, $._call)` doesn't create a hidden rule like I hoped, it inserts `_call` nodes into the final tree. Using a [anonymous alias](https://tree-sitter.github.io/tree-sitter/using-parsers#named-vs-anonymous-nodes) instead gives the tree structure we want. 

I don't know if there are any unwanted effects to setting the anonymous text here to something arbitrary ("macro call" is just a placeholder). It doesn't seem to break syntax highlighting, as far as I've tested.

Reply:
Looks good, I would go with "call" rather than "macro call" as we don't have semantic context.

Reply:
Ah, that's a good point. Crystal only allows this syntax with a macro call, but trying to enforce that in the grammar is not necessary.